src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -899,8 +899,19 @@ public override AwaitExpressionInfo GetAwaitExpressionInfo(AwaitExpressionSyntax
                 throw new ArgumentException("node.Kind==" + node.Kind());
             }
 
-            var bound = GetUpperBoundNode(node);
-            BoundAwaitableInfo awaitableInfo = (((bound as BoundExpressionStatement)?.Expression ?? bound) as BoundAwaitExpression)?.AwaitableInfo;
+            //usually the GetUpperBoundNode is sufficient, but we need to handle
+            //for example BoundExpressionStatement or BoundConversion here, too.
+            BoundAwaitExpression boundAwait = null;
+            foreach (var boundNode in GetBoundNodes(node))
```

Comment:
Did you look into just using `GetLowerBoundNode` here and assuming it's a BoundAwaitExpression in all cases?

Reply:
Yes, I actually can't imagine which node code be there beside the `BoundAwaitExpression` but to be 100% sure I would like to avoid making here any assumptions.
Usually `GetBoundNodes(node)` returns just a single node which is the one we are looking for - but as we have seen there are cases with more nodes like an additional `BoundConversion`.